heat/core/tests/test_suites/basic_test.py

```diff
@@ -21,6 +22,51 @@ def comm(self):
     def device(self):
         return BasicTest.__device
 
+    @classmethod
+    def setUpClass(cls):
+        """
+        Read the environment variable 'HEAT_TEST_USE_DEVICE' and return the requested devices.
+        Supported values
+            - cpu: Use CPU only (default)
+            - gpu: Use GPU only
+            - lcpu: GPU global, CPU local
+            - lgpu: CPU global, GPU local
+        """
+
+        envar = os.getenv("HEAT_TEST_USE_DEVICE", "cpu")
+
+        if envar == "cpu":
+            ht.use_device("cpu")
+            torch_device = ht.cpu.torch_device
+            ht_device = None
+        elif not torch.cuda.is_available():
+            raise RuntimeError(
+                "'cpu' is the only supported value for environment variable 'HEAT_TEST_USE_DEVICE' on non CUDA GPU"
+            )
+        elif envar == "gpu":
+            ht.use_device("gpu")
+            torch.cuda.set_device(torch.device(ht.gpu.torch_device))
+            torch_device = ht.gpu.torch_device
+            ht_device = None
+        elif envar == "lcpu":
+            ht.use_device("gpu")
+            torch.cuda.set_device(torch.device(ht.gpu.torch_device))
+            torch_device = ht.cpu.torch_device
+            ht_device = ht.cpu
```

Comment:
I use the ht_device for the special cases lcpu and lgpu where a tensor is created which is different from the standard device.

Reply:
not passing the device directly does not work for me

Reply:
do we really want to test this?

Reply:
A user can still want to put a specific tensor on the cpu.
It tests that a function returns the a tensor of the input device type and not the use_device one.

Reply:
A user might want to do this, but it goes against the BSP model of heat. If the user is really attempting this, he might actually want to create an entirely custom code outside heat.

Reply:
But the opposite case in 'lgpu' is okay?

Reply:
Also not, there should be GPU or CPU only for now

Reply:
Then why do we even offer the device parameter on tensor creation? :confused:

Reply:
Unless we are not meaning the same thing, allocating individual tensors on a different device should very much be allowed. For the test cases it is generally

a) not necessary to do it in this way,
b) should only be tested in a limited number of test cases in the factories function.

What I am getting here is, that you want to have different devices on different MPI ranks. This would go against the BSP model. If you want to write test cases that explicitly set the device on factory creation that is possible without the introduction of lcpu, lgpu. If you want to test whether you can allocate a DNDarray on a device different from the global one, there is no need for lcpu and lgpu, but rather something along the lines of
```
if ht.device == ht.cpu and torch.cuda.is_available:
    self.other_device = ht.gpu
else:
    self.other_device = ht.cpu
```

Reply:
As you are setting the torch device here as well. We can also remove all the ``self.device.torch_device`` calls from the test cases right?